### ISSUE TITLE
Friendlier Modifers?

### ISSUE THREAD

**Issue by Sneethe** [rpmohn/a4#16]
Is a modifier behavior that you would see on any wm possible within the confines of a terminal. 
This would mean that once your holding your modifier, multiple keyboard and or mouse actions are now possible and without having to press the modifier for each and every action. 
This would go a long for ease of use.

**Comment by rpmohn** [comment]
I would like to improve the ease of use. Can you give me an example of what you mean?

**Comment by Sneethe** [comment]
Sorry for the delay in replying. 
So with virtually every tiling window manager you generally have keybinds that can be held down to perform successive actions.  
Currently in both dvtm and a4 (to the best of my knowledge) this isn't possible. To relate this to a4, if I wish to "[Decrease the size of the zoom area](https://github.com/rpmohn/a4/blob/main/a4.1#L195)"  with 'Ctrl-a D'. 
Then the entire keybind has to be entered every time one wants to perform this action. This leads to a clunky user experience.
It would be great to hold down "Ctrl-a" / "Ctrl-g" and then have subsequent keybinds be performed as expected. 
In the above scenario that would look like: 'Ctrl-a DDDDDD'. 
Currently to do this, your keystrokes would look like: 'Ctrl-a D' &  'Ctrl-a D' &  'Ctrl-a D' &  'Ctrl-a D' &  'Ctrl-a D' &  'Ctrl-a D'.

**Comment by rpmohn** [comment]
Ah, gotcha, that makes sense and would be useful! I will look into what can be done to support something like that.

This doesn't solve it, but for those specific actions I tested out using some non-modkey bindings for them, like what's used for the S-PageUp/Down scrollback actions. If they don't conflict with your terminal emulator them maybe this will be useful for you in the meantime? I hadn't considered this before and I will certainly be using it myself now! 

```
[KeyboardActions]
M-i = zoomnum +1
M-d = zoomnum -1
M-I = zoomsize +.05
M-D = zoomsize -.05
```

**Comment by Sneethe** [comment]
Yeah setting keybinds to a terminal multiplexer is tricky, as it can't conflict with either alt and ctrl (focused application) or meta (WM/DE) keybinds.

**Comment by rpmohn** [comment]
I'm looking at implementing a `modlock` action. So, for example, have `M-C-a = modlock C-a` in your INI file, then pressing `M-C-a`  puts you in `C-a` modifier mode, and keys pressed in that mode are interpreted as having the prefix `C-a`. To exit a `modlock` mode you press `ESC`. In your example above you would type `M-C-a D D D D D D ESC`.

Do you think this would be helpful?

**Comment by rpmohn** [comment]
I've implemented this in commit e65b4d343cf36f3c4df2b8c0a093f1767783711a .
